### PR TITLE
fix highlight for extension methods

### PR DESCRIPTION
**Pull request by jkciesluk**
Highlight using presentation compiler wasn't working correctly for extension method parameter

### CODE REVIEW

File: mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala [hunk 1]
```diff
@@ -99,68 +100,103 @@ abstract class PcCollector[T](driver: InteractiveDriver, params: OffsetParams):
...
+        case Some(value) => sought
+        case None =>
+          // Fallback check for extension method parameter
+          def collectExtMethods(
+              acc: Set[(Symbol, SourcePosition)],
+              tree: untpd.Tree,
+          ) =
+            tree match
+              case ExtMethods(params :: _, _) =>
+                acc ++ params
```
Comment (tgodzik) [inline mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala:178]: params will be both type params and term params?

File: mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala [hunk 2]
```diff
@@ -185,6 +221,15 @@ abstract class PcCollector[T](driver: InteractiveDriver, params: OffsetParams):
         def soughtOrOverride(sym: Symbol) =
           sought(sym) || sym.allOverriddenSymbols.exists(sought(_))
 
+        def extensionParam(sym: Symbol) =
```
Comment (tgodzik) [inline mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala:224]: I was bit confused about the function without it.
```suggestion
        def isExtensionParam(sym: Symbol) =
```

File: tests/cross/src/test/scala/tests/highlight/Scala3DocumentHighlightSuite.scala [hunk 3]
```diff
@@ -51,4 +51,22 @@ class Scala3DocumentHighlightSuite extends BaseDocumentHighlightSuite {
...
+    """|extension (<<sb@@d>>: String)
+       |  def double = <<sbd>> + <<sbd>>
+       |  def double2 = <<sbd>> + <<sbd>>
+       |end extension
+       |""".stripMargin,
+  )
+
+  check(
+    "extension-params-ref",
+    """|extension (<<sbd>>: String)
```
Comment (tgodzik) [inline tests/cross/src/test/scala/tests/highlight/Scala3DocumentHighlightSuite.scala:65]: Could you add a test for type parameters also?

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks for finding this and great work!

File: mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala (hunk 1 above)
Comment (jkciesluk) [reply mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala]: You're right, I've changed it so it should work also when we have type params

**Comment by jkciesluk** [comment]
@tgodzik There is still an issue when we want to highlight extension type parameter like in 
```scala 
extension [<<EF>>](xs: List[EF])
      def double[CD <: EF](ys: List[EF]) = xs ++ ys
      def double2[CD <: EF](ys: List[EF]) = xs ++ ys
    end extension
```
I can't find a simple solution now, so maybe we can merge it because it's blocking https://github.com/scalameta/metals/pull/4694 and work on it in a follow-up?

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**